Doer and victim

howareyouman opened the pull request:
First try to get doer and victim from text

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -0,0 +1,107 @@
+
+
+        ObjectsAndSubjects result = new ObjectsAndSubjects();
+        Document document = new edu.stanford.nlp.simple.Document(text);
```
import?

towelenee commented on src/main/java/com/w6/nlp/Parser.java:
```diff
@@ -52,7 +55,33 @@ else if (parent.label().value().startsWith("N"))
+        
+        GetDoerAndVictim.ObjectsAndSubjects objAndSubj = GetDoerAndVictim.getSubjectAndObjectOfViolence(input);
+        
+        for(String str:objAndSubj.subjects){
```
Did you test it? It will not make sense, because text will not have it's original structure.

howareyouman replied:
not yet. It's not final version. I'd like to talk to you about it

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -0,0 +1,107 @@
+            TypedDependency typedDependency;
+            for (Object object : listObjects) {
+                typedDependency = (TypedDependency) object;
+                //System.out.println("Depdency Name" + typedDependency.dep().ner() + " :: " + "Node" + typedDependency.reln());
```
Comments should not go into production code.

howareyouman replied:
+

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -0,0 +1,107 @@
+            TreebankLanguagePack tlp = new PennTreebankLanguagePack();
+            GrammaticalStructureFactory gsf = tlp.grammaticalStructureFactory();
+            GrammaticalStructure gs = gsf.newGrammaticalStructure(tree);
+            Collection<TypedDependency> td = gs.typedDependenciesCollapsed();
```
Where do you check that sentense has a violent verb.

howareyouman replied:
Forgot it. Will correst soon.

towelenee commented on src/main/java/com/w6/nlp/Parser.java:
```diff
@@ -3,21 +3,29 @@
+    static ViolentVerbsParser violentVerbsParser;
+    
+    public Parser() throws IOException{
+        violentVerbsParser = new ViolentVerbsParser(lp);
```
why do you change static field in instance?

towelenee commented on src/main/java/com/w6/data/ObjectsAndSubjects.java:
```diff
@@ -0,0 +1,16 @@
+package com.w6.data;
```
http://docs.oracle.com/javafx/2/api/javafx/util/Pair.html

howareyouman replied:
Done

howareyouman committed:
deleted:    src/main/java/com/w6/data/ObjectsAndSubjects.java
	modified:   src/main/java/com/w6/nlp/GetDoerAndVictim.java
	modified:   src/main/java/com/w6/nlp/Parser.java
	modified:   src/test/java/ObjectAndSubjectTest.java

towelenee commented on src/test/java/ViolentVerbsTest.java:
```diff
@@ -0,0 +1,55 @@
+    public void testNoViolentVerbs() {
+        List<String> res = parser.getAllViolentVerbs("I went to school.");
+        List<String> correct = new ArrayList<String>();
+        assertTrue(res.equals(correct));
```
assertEquals

howareyouman committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into DoerAndVictim

howareyouman committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into DoerAndVictim

howareyouman committed:
Tests change and exceptions

-assertEquals
-Classes with libraries doesn't throw exceptions
